### PR TITLE
[REVIEW] Strings functionality for merge 

### PR DESCRIPTION
**Pull request by aschaffer**
This PR tracks functionality for the special case of handling strings in `merge()`.

### CODE REVIEW

File: cpp/tests/CMakeLists.txt [hunk 1]
```diff
@@ -598,6 +598,13 @@ set(MERGE_TEST_SRC
 
 ConfigureTest(MERGE_TEST "${MERGE_TEST_SRC}")
 
+###################################################################################################
+# - sorted-merge string tests ----------------------------------------------------------------------------
+
+set(MERGE_STRING_TEST_SRC
```
Comment (harrism) [inline cpp/tests/CMakeLists.txt]: Please just add this file to MERGE_TEST

File: cpp/tests/CMakeLists.txt [hunk 2]
```diff
@@ -598,6 +598,13 @@ set(MERGE_TEST_SRC
...
+  "${CMAKE_CURRENT_SOURCE_DIR}/merge/merge-str-test.cpp")
```
Comment (harrism) [inline cpp/tests/CMakeLists.txt]: Also please follow file naming conventions, and need to abbreviate. Should be `merge_string_test.cpp`

File: cpp/tests/CMakeLists.txt (hunk 1 above)
Comment (aschaffer) [reply cpp/tests/CMakeLists.txt]: Done.

File: cpp/tests/CMakeLists.txt (hunk 2 above)
Comment (aschaffer) [reply cpp/tests/CMakeLists.txt]: Done.

**Comment by jrhemstad** [comment]
@aschaffer this just needs a CHANGELOG entry and then should be good to merge.

**Pull request merged**